changelog.d/12036.misc

```diff
@@ -0,0 +1 @@
+Refactor `shared_rooms` to `mutual_rooms` (MSC2666), as per proposal changes.
```

Comment:
```suggestion
Rename `shared_rooms` to `mutual_rooms` (MSC2666), as per proposal changes.
```
I don't think this has actually changed any of the factoring of the code, so I'd just call this a rename.